python/python/lance/fragment.py

```diff
@@ -347,7 +347,7 @@ def delete(self, predicate: str) -> FragmentMetadata | None:
         >>> dataset = lance.write_dataset(tab, "dataset")
         >>> frag = dataset.get_fragment(0)
         >>> frag.delete("a > 1")
-        Fragment { id: 0, files: ..., deletion_file: Some(...), physical_rows: 3 }
+        Fragment { id: 0, files: ..., deletion_file: Some(...), physical_rows: Some(3) }
```

Comment:
in what condition that `physical_rows = None`?

Reply:
When the dataset was written prior to 0.8.0, this field is not filled in (because the field didn't exist then).

Reply:
BTW that change is more relevant to #1529, which this PR is based on. So worth reviewing that one first.